web.main.js

```diff
@@ -28,24 +32,121 @@ module.exports = function (gulpWrapper, ctx) {
+                                        noSource: true
+                                    }))
+                                    .pipe(gulp.dest(`node_modules/bundles/${fileExtension}`));
+                            }
+                            else {
+                                gulp.src(paths)
+                                    .pipe(concat(bundleElement.bundleName))
+                                    .pipe(gulp.dest(`node_modules/bundles/${fileExtension}`));
+                            }
+                        }
+                    }
+                    // CSS Files
+                    if (fileExtension && fileExtension.endsWith('css')) {
+                        if (toMinify) {
+                            gulp.src(paths)
+                                .pipe(concat(bundleElement.bundleName))
+                                .pipe(cleanCss({ inline: ['none'], level: 2 }))
+                                .pipe(gulp.dest(`node_modules/bundles/${fileExtension}`));
+                        }
+                        else {
+                            gulp.src(paths)
+                                .pipe(concat(bundleElement.bundleName))
+                                .pipe(gulp.dest(`node_modules/bundles/${fileExtension}`));
+                        }
+                    }
+                }
+            });
+        }
+
+        // Copy Assets
+        if (ctx.bundleBuilderOn &&
+            ctx.bundleBuilderAssetsConfig && ctx.bundleBuilderAssetsConfig.length > 0) {
+
+            ctx.bundleBuilderAssetsConfig.forEach(bundleElement => {
+                if (bundleElement.bundleConfigs && bundleElement.bundleConfigs.length > 0) {
+                    // Collect paths
+                    var paths = [];
+                    bundleElement.bundleConfigs.forEach(file => {
+                        var buConfig = require(ctx.baseDir + file);
+                        if (buConfig && buConfig.bundlesConfiguration && buConfig.bundlesConfiguration.length > 0) {
+                            var bundleConfig = buConfig.bundlesConfiguration.find(obj => obj.bundleAssetsTask == bundleElement.bundleAssetsTask);
+                            if (bundleConfig && bundleConfig.bundlePaths && bundleConfig.bundlePaths.length > 0) {
+                                paths = paths.concat(bundleConfig.bundlePaths)
+                            }
+                        }
+                    });
+                    return gulp.src(paths)
+                        .pipe(gulp.dest(bundleElement.bundleDestPath));
+                }
+            });
+        }
+    });
+
     /**
      * Compile typescript files
      */
     gulp.task('build', function (cb) {
-        return gulp.src('').pipe(pluginShell('\"' + process.execPath + '\" ' + typescriptCompilerPath, { cwd: ctx.baseDir }));
+        gulp.src('').pipe(pluginShell('\"' + process.execPath + '\" ' + typescriptCompilerPath, { cwd: ctx.baseDir }));
```

Comment:
With the code like this, there is no guarantee that the bundle will only start to be generated after the typescript compilation. (Although I'm not sure if that's required)
The operations should be interleaved.

Reply:
Instead of checking the bundler is ON inside 'bundle-app', we could check here and not run the task if the bundle-app is not on, or if there isn't anything to bundle.

Reply:
Correct. I did some changes to execute these tasks in sequence.
I did also the validation for bundle param ahead of time.